task/packrat/src/main.rs

```diff
@@ -408,6 +476,82 @@ impl idl::InOrderPackratImpl for ServerImpl {
             ))
         }
     }
+
+    #[cfg(not(feature = "ereport"))]
+    fn set_ereport_restart_id(
+        &mut self,
+        _: &RecvMessage,
+        _: u128,
+    ) -> Result<(), RequestError<CacheSetError>> {
+        Ok(())
+    }
+
+    #[cfg(feature = "ereport")]
+    fn set_ereport_restart_id(
+        &mut self,
+        _: &RecvMessage,
+        value: u128,
+    ) -> Result<(), RequestError<CacheSetError>> {
+        let restart_id = ereport_messages::RestartId::new(value);
+        Self::set_once(&mut self.ereport_store.restart_id, restart_id)
+            .map_err(Into::into)
+    }
+
+    #[cfg(not(feature = "ereport"))]
+    fn deliver_ereport(
+        &mut self,
+        _: &RecvMessage,
+        _: LenLimit<Leased<idol_runtime::R, [u8]>, 1024usize>,
+    ) -> Result<(), RequestError<Infallible>> {
+        // go away, we don't know how to do that
+        Err(idol_runtime::ClientError::UnknownOperation.fail())
```

Comment:
Perhaps this actually ought to just be a no-op if the feature is disabled, so that if a task tries to send an ereport on a system that doesn't support it, it just does nothing instead of getting faulted? On the other hand, we probably _do_ want any other tasks that report ereports to feature flag doing so for space-reduction reasons (and to avoid spending time CBOR-ing up a bunch of stuff that just goes to /dev/null)...

Reply:
The `net` task has a similar-ish challenge, where anyone uses `task-net-api` must agree with the `net` task itself about whether VLANs are enabled.  We could do the same here: add an `ereport` feature to `task-packrat-api`, then in the API build script confirm that it agrees with the `packrat` task itself.  That wouldn't _exactly_ prevent tasks from calling the ereport functions, but it would provide some friction.

Reply:
Ah, hm, it's actually a bit fraught: we can add a feature flag to `task-packrat-api`, but AFAICT that feature flag can't actually control whether the idol-generated functions for the ereport APIs exist or not, so it can't actually do anything useful; you can still call those functions. This isn't a problem for the net API since the feature flag doesn't actually control the presence of specific IPC interfaces.

It seems like we could go and add support to the idol codegen for only generating an IPC's client stub if a feature flag is enabled, but I'm not immediately sure how tricky that would turn out to be...

Reply:
Opened https://github.com/oxidecomputer/idolatry/issues/62 for the idea of feature-flagged IPCs in Idol codegen.